src/scripts/main.js

```diff
@@ -1,3 +1,247 @@
+    });
+  }
+
+  if (element.textContent === 'Age') {
+    rows.sort((rowA, rowB) => {
+      const A = parseInt(rowA.cells[3].innerText, 10);
+      const B = parseInt(rowB.cells[3].innerText, 10);
+
+      return sortType === 'asc' ? A - B : B - A;
+    });
+  }
+
+  if (element.textContent === 'Salary') {
+    rows.sort((rowA, rowB) => {
+      const A = parseInt(rowA.cells[4].innerText.replace('$', ''), 10);
+      const B = parseInt(rowB.cells[4].innerText.replace('$', ''), 10);
+
+      return sortType === 'asc' ? A - B : B - A;
+    });
+  }
+
+  rows.forEach((row) => tbody.appendChild(row));
+};
+
+const activateRow = () => {
+  const tbody = document.querySelector('tbody');
+
+  tbody.addEventListener('click', (e) => {
+    const clickedRow = e.target.closest('tr');
+
+    const activeRow = document.querySelector('.active');
+
+    if (activeRow && activeRow !== clickedRow) {
+      activeRow.classList.remove('active');
+    }
+
+    clickedRow.classList.toggle('active');
+  });
+};
+
+const createForm = () => {
+  const table = document.querySelector('table');
+  const form = document.createElement('form');
+  const button = document.createElement('button');
+  const inputNames = ['name', 'position', 'age', 'salary'];
+  const selectOptions = [
+    'Tokyo',
+    'Singapore',
+    'London',
+    'New York',
+    'Edinburgh',
+    'San Francisco',
+  ];
+
+  form.classList.add('new-employee-form');
+  table.insertAdjacentElement('afterend', form);
+
+  inputNames.forEach((inputName) => {
+    const label = document.createElement('label');
+    const input = document.createElement('input');
+
+    input.name = inputName;
+    input.setAttribute('data-qa', inputName);
+    input.required = true;
+
+    input.type =
+      inputName === 'age' || inputName === 'salary' ? 'number' : 'text';
+
+    label.textContent =
+      inputName.charAt(0).toUpperCase() + inputName.slice(1) + ': ';
+    label.appendChild(input);
+    form.appendChild(label);
+
+    if (inputName === 'position') {
+      const selectLabel = document.createElement('label');
+      const select = document.createElement('select');
+
+      select.name = 'office';
+      select.setAttribute('data-qa', 'office');
+      select.required = true;
+
+      selectOptions.forEach((city) => {
+        const option = document.createElement('option');
+
+        option.textContent = city;
+        select.appendChild(option);
+      });
+
+      selectLabel.textContent = 'Office: ';
+      selectLabel.appendChild(select);
+
+      form.appendChild(selectLabel);
+    }
+  });
+
+  button.type = 'submit';
+  button.textContent = 'Save to table';
+
+  form.appendChild(button);
+};
+
+const saveToTable = () => {
+  // const button = document.querySelector('button');
+
+  const form = document.querySelector('.new-employee-form');
+  const data = new FormData(form);
+  const dataToSave = {
+    name: data.get('name'),
+    position: data.get('position'),
```

Comment:
The `position` field is being fetched from the form data, but it seems like a `select` element is used for `office`. Ensure that the `position` field is correctly integrated and fetched from the form.